### PR TITLE
Make reward transaction calculation more efficent

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/transactions/v1/blockchain_txn_rewards_v1.erl [hunk 1]
```diff
@@ -234,24 +220,33 @@ to_json(Txn, _Opts) ->
...
     case blockchain:get_block(Current, Chain) of
         {error, _Reason}=Error ->
             lager:error("failed to get block ~p ~p", [_Reason, Current]),
             Error;
         {ok, Block} ->
             Transactions = blockchain_block:transactions(Block),
-            get_txns_for_epoch(Current+1, End, Chain, Txns ++ Transactions)
+            case lists:any(fun(Txn) -> blockchain_txn:type(Txn) == ?MODULE end, Transactions) of
+                true ->
+                    {error, already_existing_rewards};
```
Comment (macpie) [inline src/transactions/v1/blockchain_txn_rewards_v1.erl:243]: Was there an actual issue with this? (just curious)

File: src/transactions/v1/blockchain_txn_rewards_v1.erl [hunk 2]
```diff
@@ -234,24 +220,33 @@ to_json(Txn, _Opts) ->
...
 %% @end
 %%--------------------------------------------------------------------
--spec get_txns_for_epoch(non_neg_integer(), non_neg_integer(), blockchain:blockchain()) -> {ok, blockchain_txn:txns()}
-                                                                                           | {error, any()}.
-get_txns_for_epoch(Start, End, Chain) ->
-    get_txns_for_epoch(Start, End, Chain, []).
-
--spec get_txns_for_epoch(non_neg_integer(), non_neg_integer(),
-                         blockchain:blockchain(), blockchain_txn:txns()) -> {ok, blockchain_txn:txns()}
+-spec get_rewards_for_epoch(non_neg_integer(), non_neg_integer(),
```
Comment (macpie) [inline src/transactions/v1/blockchain_txn_rewards_v1.erl:223]: maybe rename this to `get_rewards_for_epoch` as it only return POC rewards?

File: src/transactions/v1/blockchain_txn_rewards_v1.erl (hunk 1 above)
Comment (Vagabond) [reply src/transactions/v1/blockchain_txn_rewards_v1.erl:243]: I was just preserving the existing check we had

Comment (macpie) [reply src/transactions/v1/blockchain_txn_rewards_v1.erl:243]: oh yes did not see it

File: src/transactions/v1/blockchain_txn_rewards_v1.erl (hunk 2 above)
Comment (Vagabond) [reply src/transactions/v1/blockchain_txn_rewards_v1.erl:223]: I was going to calculate DC rewards in here as well.

**Pull request merged**